### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by goldenuni**

### CODE REVIEW

File: app/knight.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, info: dict) -> None:
```
Comment (olena-hul) [inline app/knight.py:6]: Use better variable name

File: app/knight.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            for part in info["armour"]:
+                self.protection += part["protection"]
+
+        if info["weapon"] is not None:
+            self.power += info["weapon"]["power"]
+
+        if info["potion"] is not None:
+            self.power += info["potion"]["effect"].get("power", 0)
+            self.hp += info["potion"]["effect"].get("hp", 0)
+            self.protection += info["potion"]["effect"].get("protection", 0)
```
Comment (olena-hul) [inline app/knight.py:22]: Use `getattr` and `setattr` and do this in a loop

File: app/knight.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class Knight:
+
+    def __init__(self, info: dict) -> None:
+        self.name = info["name"]
+        self.power = info["power"]
+        self.hp = info["hp"]
+        self.protection = 0
+
+        if info["armour"] != []:
```
Comment (olena-hul) [inline app/knight.py:12]:
```suggestion
        if info.get("armour")

```

File: app/knight.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            for part in info["armour"]:
+                self.protection += part["protection"]
+
+        if info["weapon"] is not None:
```
Comment (olena-hul) [inline app/knight.py:16]: The same here

File: app/knight.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return f"{self.name} | HP: {self.hp} | Power: {self.power} " \
+               f"| Protection: {self.protection}"
+
+    @classmethod
+    def battle(cls, first_knight: Knight, second_knight: Knight) -> None:
+        first_knight.hp -= second_knight.power - first_knight.protection
+        second_knight.hp -= first_knight.power - second_knight.protection
+
+        if first_knight.hp <= 0:
+            first_knight.hp = 0
```
Comment (olena-hul) [inline app/knight.py:34]: Move this to a separate method

File: app/main.py [hunk 6]
```diff
@@ -1,214 +1,19 @@
...
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    for knight in knights_config.values():
+        tournament_members.append(Knight(knight))
```
Comment (olena-hul) [inline app/main.py:9]: You can use list comprehension for this

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/knight.py (hunk 1 above)
Comment (goldenuni) [reply app/knight.py:6]: @olena-hul better variable name for info (parameter in __init__) ?

Comment (olena-hul) [reply app/knight.py:6]: Yes, name it knight, for example

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**